### ISSUE TITLE
Best checkpoint selection - OOD test set leakage

### ISSUE THREAD

**Issue by backpropagate404** [robometer/robometer#30]
Hello,

I am afraid the current codebase settings selects the best checkpoint on the OOD test set, meaning the "held-out OOD set" is actually used for selecting the best model.

 In the current code guidelines, the training command shows:

```
**Train on RBM-1M in-distribution and evaluate on RBM-1M-OOD**
First, modify `robometer/configs/config.yaml`'s `wandb_entity` flag to your WandB entity. To disable WandB logging, remove "wandb" from the `log_to` list in the config yaml file.
See more flags in the config file (e.g., batch size, learning rates, etc.)

uv run accelerate launch --config_file robometer/configs/distributed/fsdp.yaml --num_processes=N_GPUS_YOU_HAVE train.py \
  data.train_datasets=[rbm-1m-id] \
  data.eval_datasets=[rbm-1m-ood] \
  data.max_frames=8 \
  model.train_progress_head=true \
  model.train_preference_head=true \
  training.max_steps=15000 \
  custom_eval.reward_alignment=[rbm-1m-ood] \
  custom_eval.policy_ranking=[rbm-1m-ood] \
  custom_eval.confusion_matrix=[rbm-1m-ood] \
  logging.save_best.metric_names=[eval_p_rank/kendall_last_utd_so101_clean_top,eval_p_rank/kendall_last_usc_xarm,eval_p_rank/kendall_last_usc_franka,eval_p_rank/kendall_last_rfm_new_mit_franka_nowrist,eval_p_rank/kendall_last_usc_trossen] \
  logging.save_best.greater_is_better=[True,True,True,True,True]
```

Selection is wired to OOD eval performance, not RBM-1M-ID validation.

For reproducing the paper results, should we use the command as is?

Thank you!

**Comment by aliang8** [comment]
Hi,

Thank you for the comment! We do not use this for the official model training. You should train the model to completion and use the final checkpoint. Those lines were meant to be used for debugging purposes. We will update the training command to reflect this!
